libultraship/libultraship/BinaryReader.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <string>
+#include <memory>
+#include "BitConverter.h"
+#include "Vec2f.h"
+#include "Vec3f.h"
+#include "Vec3s.h"
+#include "Color3b.h"
+
+class BinaryReader;
+
+namespace Ship {
+
+enum class SeekOffsetType
+{
+    Start,
+    Current,
+    End
+};
+
+class BinaryReader
+{
+public:
+    BinaryReader(char* buffer, uint32_t size);
+
+    void SetEndianness(Endianness endianness);
+    Endianness GetEndianness() const;
+
+    void Seek(int32_t offset, SeekOffsetType seekType);
+    uint32_t GetBaseAddress();
+
+    void Read(int32_t length);
+    void Read(char *buffer, int32_t length);
+    char ReadChar();
+    int8_t ReadByte();
```

Comment:
```suggestion
    int8_t ReadInt8();
```
For consistency

Reply:
Same as the Single one, done to mimic the previous ABI.

Reply:
Should be fixed in recent commit.